relay-sampling/src/evaluation.rs

```diff
@@ -95,139 +114,193 @@ pub fn merge_configs_and_match(
     // The determination of the sampling mode occurs with the following priority:
     // 1. Non-root project sampling mode
     // 2. Root project sampling mode
-    let Some(primary_config) = sampling_config.or(root_sampling_config) else {
-        relay_log::error!("cannot sample without at least one sampling config");
```

Comment:
previously this was unreachable as a lack of either config would have resulted in no rules, thereby no matches, which would have returned early here: https://github.com/getsentry/relay/blob/cb88f1227eefdad162430f232557b2f98db58f9d/relay-sampling/src/evaluation.rs#L88 

I changed it from error to info so not to fail all the CI tests that dont pass in any configs.